src/Field/Square.java

```diff
@@ -1,8 +1,27 @@
 package Field;
 
-public interface Square {
+public abstract class Square {
 
-    boolean isAMine();
+    private boolean isRevealed = false;
+    private int hintCount = 0;
 
-    boolean isRevealed();
+    boolean isAMine() {
+        return false;
+    }
+
+    public boolean isRevealed() {
+        return isRevealed;
+    }
+
+    public void setRevealed() {
```

Comment:
Maybe just call this method `reveal`, since it always sets `isRevealed` to true. `setReveal` implies that this is going to allow you to set the value.